File: src/test/java/com/ddd/trainings/idea/domain/IdeaNameVOTest.java
package com.ddd.trainings.idea.domain;

import static org.assertj.core.api.Assertions.assertThat;

import com.ddd.trainings.TrainingTests;
import org.junit.jupiter.api.Test;

class IdeaNameVOTest extends TrainingTests {

    @Test
    void from_createsIdeaName_onValidName() {
        // given
        String validName = "This is a valid idea name that meets the length requirements";

        // when
        IdeaNameVO ideaName = IdeaNameVO.from(validName);

        // then
        assertThat(ideaName).isNotNull();
        assertThat(ideaName.getNameValue()).isEqualTo(validName);
    }
    
    @Test
    void from_throwsException_onInvalidLengthName() {
        //TODO: Implement unit tests
    }

    @Test
    void getKeywords_returnsKeywords_onValidName() {
        // given
        String name = "Java Spring Boot Microservices";
        IdeaNameVO ideaName = IdeaNameVO.from(name);

        // when
        String[] keywords = ideaName.getKeywords();

        // then
        assertThat(keywords)
            .isNotNull()
            .hasSize(4)
            .containsExactly("Java", "Spring", "Boot", "Microservices");
    }
}

File: src/Ddd.Tests/Trainings/Idea/Domain/IdeaNameVOTest.cs
using Ddd.Tests.Trainings.Tests;
using Ddd.Trainings.Idea.Domain;
using Xunit;

namespace Ddd.Tests.Trainings.Idea.Domain.Tests;

public class IdeaNameVOTest : TrainingTests
{
  

  [Fact]
  public void From_CreatesIdeaName_OnValidName()
  {
    // given
    string validName = "This is a valid idea name that meets the length requirements";

    // when
    IdeaNameVO ideaName = IdeaNameVO.From(validName);

    // then
    Assert.NotNull(ideaName);
    Assert.Equal(validName, ideaName.NameValue);
  }

  [Fact]
  public void From_ThrowsException_OnInvalidLengthName()
  {
    //TODO: Implement unit tests
  }

  [Fact]
  public void GetKeywords_ReturnsKeywords_OnValidName()
  {
    // given
    string name = "Java Spring Boot Microservices";
    string[] ExpectedKeywords = ["Java", "Spring", "Boot", "Microservices"];
    IdeaNameVO ideaName = IdeaNameVO.From(name);

    // when
    string[] keywords = ideaName.Keywords;

    // then
    Assert.NotNull(keywords);
    Assert.Equal(4, keywords.Length);
    Assert.Equal(ExpectedKeywords, keywords);
  }
}